👞 Footer Layout

jasangill1 opened the pull request:
## About 

- Want to know what to do for the on hover, current I just took the hover that Joe made for the buttons under the hero title. 
- Maked a very basic list for the footer links started the status button. 
- Let me know oh how you would wan that ehover effects

asim-shrestha commented on next/src/components/landing/FooterLinks.tsx:
```diff
@@ -0,0 +1,30 @@
+    <div className="flex cursor-pointer justify-between space-x-4">
+      <a
+        href="#https://www.ycombinator.com/companies/reworkd/jobs"
+        className="rounded-full bg-transparent text-white/60 transition duration-200 ease-in-out hover:bg-neutral-900 hover:text-neutral-100 focus-visible:bg-neutral-900 focus-visible:outline-none focus-visible:ring-4 focus-visible:ring-neutral-500"
```
We shouldn't be duplicating these for each button. Lets extract this into a common component and use for each of the links

asim-shrestha commented on next/src/components/landing/Hero.tsx:
```diff
@@ -82,8 +83,12 @@ const Hero = () => {
         initialY={50}
         duration={3}
         className="absolute bottom-10 right-0 z-10 w-screen justify-center"
+        s
```
This is making the deployment fail

jasangill1 commented:
![image](https://github.com/reworkd/AgentGPT/assets/56135840/78a36c21-d317-487b-a1a7-fab0a889584d)

asim-shrestha commented:
A bit sus on mobile

asim-shrestha commented on next/src/components/landing/FooterLinks.tsx:
```diff
@@ -0,0 +1,32 @@
+  <a
+    href={href}
+    className={clsx(
+      "w-full rounded-full bg-transparent text-white/20 ",
```
lets use a text-neutral by default. Currently its too dark to see when not hovered

asim-shrestha replied:
Also, we can just pass in the `Status` stuff as children as well so that it can be

asim-shrestha commented on next/src/components/landing/FooterLinks.tsx:
```diff
@@ -0,0 +1,32 @@
+import clsx from "clsx";
+
+const FooterLink = ({ href, children }) => (
```
Missing typescript types. Your editor should be complaining about this

jasangill1 replied:
typing it daddy

awtkns commented on next/src/components/landing/FooterLinks.tsx:
```diff
@@ -0,0 +1,32 @@
+    className={clsx(
+      "w-full rounded-full bg-transparent text-white/20 ",
+      "transition duration-200 ease-in-out hover:text-white/90",
+      " focus:outline-none focus:ring-2 focus:ring-white/60 focus:ring-offset-2 focus:ring-offset-transparent"
```
```suggestion
      "focus:outline-none focus:ring-2 focus:ring-white/60 focus:ring-offset-2 focus:ring-offset-transparent"
```

awtkns replied:
nit: Of the mind we shouldn't be using clsx unless there are variables. Instead you can just setup line wrapping.

CLSX is worse for performance and makes it so tw classes are no longer sorted. 

Not a huge issue just something to keep in mind for the future

awtkns commented on next/src/components/landing/FooterLinks.tsx:
```diff
@@ -0,0 +1,32 @@
+  <a
+    href={href}
+    className={clsx(
+      "w-full rounded-full bg-transparent text-white/20 ",
```
```suggestion
      "w-full rounded-full bg-transparent text-white/20",
```

awtkns commented:
<img width="369" alt="Screenshot 2023-07-11 at 3 49 32 PM" src="https://github.com/reworkd/AgentGPT/assets/32209255/10c6490f-d205-48c9-8bdf-938d3987bb28">

The alignment of these need to be consistent

awtkns commented on next/src/components/landing/Hero.tsx:
```diff
@@ -83,7 +84,10 @@ const Hero = () => {
         className="absolute bottom-10 right-0 z-10 w-screen justify-center"
       >
-        <Backing />
+        <div className="flex justify-between px-4 md:px-8 lg:px-16">
```
Similar to the figma design we also need to add the copyright notice

jasangill1 replied:
yes daddy

awtkns reviewed (changes requested):
Please address @asim-shrestha's comments and see comments about copyright / alignment of footer items

jasangill1 commented:
![image](https://github.com/reworkd/AgentGPT/assets/56135840/56b3c396-b77e-44b6-9f0c-54ad14b18866)

![image](https://github.com/reworkd/AgentGPT/assets/56135840/dad5b4ec-adfe-43f8-85d8-a47ac89b3cbe)


adjusted the things in the comments above !! fixed the styling to be more like Figma
